pysmiles/smiles_helper.py

```diff
@@ -349,8 +349,10 @@ def bonds_missing(mol, node_idx, use_order=True):
     bonds += mol.nodes[node_idx].get('hcount', 0)
 
     val = valence(mol.nodes[node_idx])
-    val = [v for v in val if v >= bonds] or [0]
-    return int(max(val[0] - bonds, 0))
+    if not val:
+        return 0
+    val = [v for v in val if v >= bonds] or val[-1:]
```

Comment:
why val[-1:]? I fail to see the case

Reply:
If there are more bonds than valence, take the largest possible valence to determine how wrong it is.